fsspec/spec.py

```diff
@@ -649,6 +649,50 @@ def isfile(self, path):
         except:  # noqa: E722
             return False
 
+    def read_text(self, path, encoding=None, errors=None, newline=None, **kwargs):
```

Comment:
A `read_text`/`write_text` in 2022 should probably just default to `utf-8`. In Python 3.10, `Path.read_text()` raises `EncodingWarning` if `-X warn_default_encoding` is set. Also, Python recommends to set encoding to "utf-8" by default for new APIs:

> please consider using UTF-8 by default (i.e. encoding="utf-8") for new APIs.

See https://docs.python.org/3/library/io.html#text-encoding.

Reply:
I didn't know that. Is this also true for builtins.open or TextIOWrapper? Certainly, utf8 is the only encoding I've come across, but people on windows still seem to have latin1 and others.

Reply:
> Is this also true for builtins.open or TextIOWrapper?

No, there's a new `io.text_encoding()` that has to be opt-in by the users of `open()` and `TextIOWrapper()` for this. That's what `Path().read_text()` does. See https://docs.python.org/3/library/io.html#opt-in-encodingwarning.

> If you are providing an API that uses [open()](https://docs.python.org/3/library/functions.html#open) or [TextIOWrapper](https://docs.python.org/3/library/io.html#io.TextIOWrapper) and passes encoding=None as a parameter, you can use [text_encoding()](https://docs.python.org/3/library/io.html#io.text_encoding) so that callers of the API will emit an [EncodingWarning](https://docs.python.org/3/library/exceptions.html#EncodingWarning) if they don’t pass an encoding. However, please consider using UTF-8 by default (i.e. encoding="utf-8") for new APIs.

Reply:
Reading the PEP again, yes it also applies to `builtins.open()` and `io.TextIOWrapper`. 🤦🏼 

> When the flag is set, io.TextIOWrapper(), open() and other modules using them will emit EncodingWarning when the encoding argument is omitted.